packages/core/src/applicationcomposer/messageHandlers/initMessageHandler.ts

```diff
@@ -5,19 +5,24 @@
 
 import path from 'path'
 import { InitResponseMessage, MessageType, WebviewContext, Command } from '../types'
-import { AuthUtil } from '../../codewhisperer/util/authUtil'
+import { getAmazonqApi } from '../../amazonq/extApi'
 
 export async function initMessageHandler(context: WebviewContext) {
     const filePath = context.defaultTemplatePath
-    const authState = await AuthUtil.instance.getChatAuthState()
+    const amazonqApi = await getAmazonqApi()
+    let isConnectedToCodeWhisperer = false
+    if (amazonqApi) {
+        const authState = await amazonqApi.authApi.getChatAuthState()
+        isConnectedToCodeWhisperer =
+            authState.codewhispererChat === 'connected' || authState.codewhispererChat === 'expired'
+    }
```

Comment:
since this pattern is appearing in a few places i will suggest:

```suggestion
    const amazonqApi = await getAmazonqApi()
    const authState = await amazonqApi?.authApi.getChatAuthState()
    const isConnectedToAmazonQ = authState?.codewhispererChat === 'connected' || authState?.codewhispererChat === 'expired'
```

note: `await undefined` is legal